extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoHandler.java

```diff
@@ -338,44 +351,75 @@ public void onValueReceived(String variable, String value, String service) {
 
     private synchronized void onSubscription() {
         if (service.isRegistered(this)) {
-            logger.debug("Setting up WeMo GENA subscription for '{}'", this);
+            logger.debug("Checking WeMo GENA subscription for '{}'", this);
 
             ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+            String subscription = null;
 
             if (thingTypeUID.equals(THING_TYPE_INSIGHT)) {
-                service.addSubscription(this, "insight1", SUBSCRIPTION_DURATION);
+                subscription = "insight1";
+
             } else {
+                subscription = "basicevent1";
                 service.addSubscription(this, "basicevent1", SUBSCRIPTION_DURATION);
             }
+
+            if ((subscriptionState.get(subscription) == null) || !subscriptionState.get(subscription).booleanValue()) {
+                logger.debug("Setting up GENA subscription {}: Subscribing to service {}...", getUDN(), subscription);
+                service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
+                subscriptionState.put(subscription, true);
+            }
```

Comment:
I don't see any check of subscriptionState before adding a new subscription. Am I missing something?